### DISCUSSION TITLE
Line pruning?

### DISCUSSION THREAD

**Discussion by kyllerss** [maplibre/maplibre-gl-js#4201] [Q&A] [answered]
I am rendering a path of a swim using maplibre-gl. Perhaps my approach is wrong, but I'm trying to color the path based on the pace of the swimmer. The way I chose to do this was to render a separate line (however short it may be) with the given pace color:

![Screenshot_20240529-190303.png](https://github.com/maplibre/maplibre-gl-js/assets/65538/dd2b0c9a-2364-4cf6-8c35-b52b5b41f014)

What I found was that maplibre will remove some of these lines as a display optimization. When I zoom in, the missing segments appear:

![Screenshot_20240529-190340.png](https://github.com/maplibre/maplibre-gl-js/assets/65538/3c31b654-e97c-40f2-bd69-8d2de5dac133)

So my question is two-fold:

- Is there a way of controlling how aggressive this visual optimization is?
- Is there a better way of doing this?

**Comment by HarelM** [comment, marked as answer]
Without seeing your code it's very hard to tell what's wrong. 
I have a similar feature where I show the slope of a route using color where blue is downhill and red is uphill. 
The code is here: 
https://github.com/IsraelHikingMap/Site/blob/825d8336e24601b9c0b1f18e79e16c87c7989a5e/IsraelHiking.Web/src/application/components/route-statistics.component.ts#L1028

I don't have any gaps, I'm drawing a single line and color it with "stops" by converting line progress to the relevant color up front.

**Comment by kyllerss** [comment]
Thanks, @HarelM! That's a good suggestion. I had been looking at stops as an alternative, but it seemed to me that it had to be a linear progression. Thank you for the tip!
